### DISCUSSION TITLE
Query with SHACL

### DISCUSSION THREAD

**Discussion by thadguidry** [ArcadeData/arcadedb#370] [Open Roadmap]
@lvca Will SHACL eventually be available as a query language for graph queries?

Use Case:  I would like to use SHACL for data quality and validation (ensuring uniqueness that all Product nodes must have no more than a 1:1 relationship with a Manufacturer node.  If a Product node has 1:n with Manufacturer nodes, then likely the Product node is claiming to be made by 2 or more different Manufacturers and in my DB model that would not be allowed (in real world it typically is) and a new unique Product node would need to be made for each Manufacturer.

Use Case: I would like to easily find nodes based on certain attributes and define them as a set of must-have constraints where some attributes are themselves a derived shape expression in SHACL (I.E. a User Defined Type in some systems).  Find ProductType: Olive (from Manufacturers in Baltic states only where we have existing Sales relationship) that are FoodType: Certified Organic

**Comment by lvca** [comment]
@thadguidry sorry, no plan to support SHACL in the short term, unless we get a sponsor for it or some volunteer that wants to contribute.

**Reply by thadguidry** [reply]
Then I guess I can get some semblance of constraints with GraphQL interfaces, union types, and [fragments on composite types](https://spec.graphql.org/draft/#sec-Fragments-On-Composite-Types).
